pages/page.ts

```diff
@@ -0,0 +1,46 @@
+import { Locator, type Page } from "@playwright/test";
+
+abstract class BasePage {
+  readonly page: Page;
+
+  constructor(page: Page) {
+    this.page = page;
+  }
+
+  async goto(url: string) {
+    await this.page.goto(url);
+  }
+
+  async getElement(element: string): Promise<Locator> {
+    return this.page.locator(element);
+  }
+
+  async getElementByTestID(dataTestId: string): Promise<Locator> {
+    return this.page.getByTestId(dataTestId);
+  }
+
+  async getNestedElement(parent: string, child: string): Promise<Locator> {
+    return this.page.locator(parent).locator(child);
+  }
+
+  async clickElement(element: string) {
+    await (await this.getElementByTestID(element)).click();
```

Comment:
pass Locator object to this kind of methods instead of string, you would not always click element byTestId